source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -95,93 +101,177 @@ namespace aspect
         // which usually is not a square matrix. Therefore, we find the
         // least squares solution of Ac=r by solving the reduced QR factorization
         // Ac = QRc = b -> Q^TQRc = Rc =Q^Tb
-        dealii::ImplicitQR<dealii::Vector<double>> qr;
-
         // A is a std::vector of Vectors(which are it's columns) so that we create what the ImplicitQR
         // class needs.
-        std::vector<dealii::Vector<double>> A(n_matrix_columns, dealii::Vector<double>(n_particles));
+        std::vector<Vector<double>> A(n_matrix_columns, Vector<double>(n_particles));
         std::vector<Vector<double>> b(n_particle_properties, Vector<double>(n_particles));
-        std::vector<Vector<double>> QTb(n_particle_properties, Vector<double>(n_matrix_columns));
-        std::vector<Vector<double>> c(n_particle_properties, Vector<double>(n_matrix_columns));
-        for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
-          if (selected_properties[property_index])
-            b[property_index] = 0;
 
-        unsigned int particle_index = 0;
-        const double cell_diameter = found_cell->diameter();
+        AssertThrow(n_particles >= n_matrix_columns,
```

Comment:
Move this assert further up. Also, why do you assert instead of using the fallback interpolator?